src/evox/algorithms/mo/tdea.py

```diff
@@ -0,0 +1,176 @@
+import jax
+import jax.numpy as jnp
+
+from evox.operators import mutation, crossover, sampling
+from evox.operators import non_dominated_sort
+from evox.utils import cos_dist
+from evox import Algorithm, State, jit_class
+from functools import partial
+
+
+@jax.jit
+def theta_nd_sort(obj, w, mask):
+    # theta non-dominated sort
+    n = len(obj)
+    nw = len(w)
+
+    norm_p = jnp.sqrt(jnp.sum(obj**2, axis=1, keepdims=True))
+    cosine = cos_dist(obj, w)
+    cosine = jnp.clip(cosine, -1, 1)
+    d1 = jnp.tile(norm_p, (1, nw)) * cosine
+    d2 = jnp.tile(norm_p, (1, nw)) * jnp.sqrt(1 - cosine**2)
+
+    d_class = jnp.argmin(d2, axis=1)
+    d_class = jnp.where(mask, d_class, jnp.inf)
+
+    theta = jnp.zeros((nw,)) + 5
+    theta = jnp.where(jnp.sum(w > 1e-4, axis=1) == 1, 1e6, theta)
+    t_rank = jnp.zeros((n,), dtype=int)
+
+    t_list = jnp.arange(1, n + 1)
+
+    def loop_body(i, val):
+        t_front_no = val
+        loop_mask = d_class == i
+        d = jnp.where(loop_mask, d1[:, i] + theta[i] * d2[:, i], jnp.inf)
+        rank = jnp.argsort(d)
+        tmp = jnp.where(loop_mask[rank], t_list, t_front_no[rank])
+        t_front_no = t_front_no.at[rank].set(tmp)
+
+        return t_front_no
+
+    t_rank = jax.lax.fori_loop(0, nw, loop_body, t_rank)
+    t_rank = jnp.where(mask, t_rank, jnp.inf)
+
+    return t_rank
+
+
+@partial(jax.jit, static_argnums=3)
+def environmental_selection(pop, obj, w, n, z, z_nad):
+
+    n_merge, m = jnp.shape(obj)
+    rank = non_dominated_sort(obj)
+    order = jnp.argsort(rank)
+    worst_rank = rank[order[n]]
+    mask = rank <= worst_rank
+
+    z = jnp.minimum(z, jnp.min(obj, axis=0))
+
+    w1 = jnp.zeros((m, m)) + 1e-6
+    w1 = jnp.where(jnp.eye(m), 1, w1)
+    asf = jax.vmap(
+        lambda x, y: jnp.max(jnp.abs((x - z) / (z_nad - z)) / y, axis=1),
+        in_axes=(None, 0),
+        out_axes=1,
+    )(obj, w1)
+
+    extreme = jnp.argmin(asf, axis=0)
+    hyper_plane = jnp.linalg.solve(
+        obj[extreme, :] - jnp.tile(z, (m, 1)), jnp.ones((m, 1))
+    )
+    a = z + 1 / jnp.squeeze(hyper_plane)
+
+    a = jax.lax.cond(
+        jnp.any(jnp.isnan(a)) | jnp.any(a <= z),
+        lambda _: jnp.max(obj, axis=0),
+        lambda x: x,
+        a,
+    )
+    z_nad = a
+
+    norm_obj = (obj - jnp.tile(z, (n_merge, 1))) / jnp.tile(z_nad - z, (n_merge, 1))
+
+    t_rank = theta_nd_sort(norm_obj, w, mask)
+    combined_order = jnp.lexsort((t_rank, rank))[:n]
+
+    return pop[combined_order], obj[combined_order], z, z_nad
+
+
+@jit_class
+class tDEA(Algorithm):
```

Comment:
```suggestion
class TDEA(Algorithm):
```
Follow Python's naming convention